src/pipeline/src/etl/processor/select.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use ahash::{HashSet, HashSetExt};
+use snafu::OptionExt;
+
+use crate::error::{Error, InvalidInputSnafu, KeyMustBeStringSnafu, Result};
+use crate::etl::field::Fields;
+use crate::etl::processor::{
+    yaml_new_field, yaml_new_fields, yaml_string, FIELDS_NAME, FIELD_NAME, TYPE_NAME,
+};
+use crate::{PipelineMap, Processor};
+
+pub(crate) const PROCESSOR_SELECT: &str = "select";
+const RETAIN: &str = "retain";
+const DISASSOCIATE: &str = "disassociate";
```

Comment:
Asked AI to give us some alternative names for this:

Suggested Pairings:

    KEEP and DROP
    INCLUDE and EXCLUDE
    PRESERVE and REMOVE
    SELECT and DISCARD
    SAVE and OMIT

Reply:
Sure